app/scripts/HiGlassComponent.js

```diff
@@ -4415,6 +4422,33 @@ class HiGlassComponent extends React.Component {
     );
   }
 
+  /**
+   * Handle trackDimensionsModified events
+   * settings.trackId = id of the track
+   * settings.height = new height of the track or undefined if current height should remain
+   * settings.width = new width of the track or undefined if current width should remain
+   */
+  trackDimensionsModifiedHandler(settings) {
```

Comment:
Track uids are not guaranteed to be globally unique. You need a view / track uid pair to uniquely identify a track.

Reply:
Oh yes, of course. Thanks. Is there a simple way to get the current viewId from within a (plugin) track?

Reply:
Unfortunately, I don't think there is. You could perhaps add it to the context here: https://github.com/higlass/higlass/blob/9ceffd59cd75dff315cf67e33185da185b8c7419/app/scripts/TrackRenderer.js#L1491 but I'm not sure that it's even present in TrackRenderer.

Another possibility is to pass a callback into the track object that lets it change its size. See the viewport projection tracks as an example of this:

https://github.com/higlass/higlass/blob/9ceffd59cd75dff315cf67e33185da185b8c7419/app/scripts/TrackRenderer.js#L1621

Reply:
Thanks. It turns out the TrackRenderer knows about it. It just wasn't used so far.